### PR TITLE
refactor(fs): Don't proxy `writeFile` function

### PR DESCRIPTION
**Pull request by zharinov**
## Context

- Ref: #16125

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/global/index.spec.ts [hunk 1]
```diff
@@ -176,7 +180,7 @@ describe('workers/global/index', () => {
         endpoint: 'https://github.com/',
         writeDiscoveredRepos: '/tmp/renovate-output.json',
       });
-      fs.writeFile.mockResolvedValueOnce();
+      fs.writeFile.mockResolvedValueOnce(null as never);
```
Comment (viceice) [inline lib/workers/global/index.spec.ts:183]: why this, write file is a void function

**Review by viceice — COMMENTED** [review]
do you plan to completely remove the proxies?

**Comment by zharinov** [comment]
> do you plan to completely remove the proxies?

Planning to make them limited to either `localDir` or `cacheDir`

File: lib/workers/global/index.spec.ts (hunk 1 above)
Comment (viceice) [reply lib/workers/global/index.spec.ts:183]: does write file have a return value?

**Review by rarkins — DISMISSED** [review]

File: lib/workers/global/index.spec.ts [hunk 2]
```diff
@@ -15,6 +16,9 @@ jest.mock('../repository');
 jest.mock('../../util/fs');
 jest.mock('../../config/presets');
 
+jest.mock('fs-extra');
+const fs: jest.Mocked<typeof _fs> = _fs as never;
```
Comment (viceice) [inline lib/workers/global/index.spec.ts:20]:
```suggestion
const fs = mocked(_fs);
```

**Review by viceice — COMMENTED** [review]
also just a small thing 🙃

**Review by viceice — APPROVED** [review]

**Pull request merged**